OpenHome/NetworkAdapterList.cpp

```diff
@@ -314,101 +314,107 @@ TBool NetworkAdapterList::CompareSubnets(NetworkAdapter* aI, NetworkAdapter* aJ)
 void NetworkAdapterList::HandleInterfaceListChanged()
 {
     static const char* kRemovedAdapterCookie = "RemovedAdapter";
-    iListLock.Wait();
```

Comment:
This wait wasn't unwound, if the code to detect the network through.  That left a mutex held by a non-existant thread.

The change here is to use autoMutex, but some variables and scopes had to be moved around to get it to work.